src/scripts/main.js

```diff
@@ -1 +1,55 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', () => {
+    resolve(`First promise was resolved`);
+  });
+  setTimeout(() => reject(new Error(`First promise was rejected`)), 3000);
+});
+const secondPromise = new Promise((resolve) => {
+  document.addEventListener('click', () => {
+    resolve(`Second promise was resolved`);
+  });
+
+  document.addEventListener('contextmenu', () => {
+    resolve(`Second promise was resolved`);
+  });
+});
+const thirdPromise = new Promise((resolve) => {
+  let clicksCount = 0;
+
+  document.addEventListener('click', handleThirdEvents);
+
+  document.addEventListener('contextmenu', handleThirdEvents);
```

Comment:
When utilizing the context menu in Event, be cautious, as it might be a trap. If you press Shift + F10 in the browser task, you'll notice that the operation is executed. Therefore, the intention is to either right-click or left-click.

You can use: 
`
addEventListener('mouseup', (e) => {
  if (e.button === 0 || e.button === 2)
`